web/controllers/api/docs_controller.ex

```diff
@@ -6,21 +6,20 @@ defmodule HexWeb.API.DocsController do
   @uncompressed_max_size 64 * 1024 * 1024
 
   def show(conn, %{"name" => name, "version" => version}) do
-    if (package = HexWeb.Repo.get_by(Package, name: name)) &&
-       (release = HexWeb.Repo.get_by(assoc(package, :releases), version: version)) do
-      if release.has_docs do
-        redirect conn, external: HexWeb.Utils.docs_tarball_url(package, release)
-      else
-        not_found(conn)
-      end
+    package = HexWeb.Repo.get_by(Package, name: name)
```

Comment:
This code could likely be moved to a function. At least the package + release fetching.